query/src/test/java/org/infinispan/query/distributed/DistributedReindexTest.java

```diff
@@ -0,0 +1,75 @@
+package org.infinispan.query.distributed;
+
+import static org.testng.AssertJUnit.assertEquals;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.lucene.search.Query;
+import org.infinispan.Cache;
+import org.infinispan.context.Flag;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.query.CacheQuery;
+import org.infinispan.query.Search;
+import org.infinispan.query.SearchManager;
+import org.infinispan.query.test.ReindexTestObject;
+import org.infinispan.test.MultipleCacheManagersTest;
+import org.infinispan.test.fwk.TestCacheManagerFactory;
+import org.testng.annotations.Test;
+
+/**
+ * Created by Seto on 2017/3/22.
+ */
+@Test(groups = "functional", testName = "query.distributed.DistributedReindexTest")
+public class DistributedReindexTest extends MultipleCacheManagersTest {
```

Comment:
We already have a test for the distributed ```MassIndexer```, called ```org.infinispan.query.distributed.DistributedMassIndexingTest```, no need to create another. 
Please add a method to that class with the ```reindex``` method.